app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/fragments/CharacterFragmentDialog.kt

```diff
@@ -0,0 +1,54 @@
+package com.puzzlebench.clean_marvel_kotlin.presentation.fragments
+
+
+import android.app.Dialog
+import android.os.Bundle
+import android.support.v4.app.DialogFragment
+import android.support.v4.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.Window
+import com.puzzlebench.clean_marvel_kotlin.R
+import com.puzzlebench.clean_marvel_kotlin.presentation.MainActivity
+import com.puzzlebench.clean_marvel_kotlin.presentation.fragments.mvp.FragmentDialogModel
+import com.puzzlebench.clean_marvel_kotlin.presentation.fragments.mvp.FragmentDialogPresenter
+import com.puzzlebench.clean_marvel_kotlin.presentation.fragments.mvp.FragmentDialogView
+import com.puzzlebench.cmk.data.service.CharacterServicesImpl
+import com.puzzlebench.cmk.domain.model.Character
+import com.puzzlebench.cmk.domain.usecase.GetSingleCharacterServiceUseCase
+
+/**
+ * A simple [Fragment] subclass.
+ */
+class CharacterFragmentDialog : DialogFragment() {
+
+    private val singleCharacterUseCase = GetSingleCharacterServiceUseCase(CharacterServicesImpl(), character.id)
+    private val presenter = FragmentDialogPresenter(FragmentDialogView(mainActivity), FragmentDialogModel(singleCharacterUseCase))
+
+    companion object {
+        private lateinit var character: Character
+        private lateinit var mainActivity: MainActivity
+        fun newInstance(character: Character, activity: MainActivity?): CharacterFragmentDialog {
+            this.character = character
+            mainActivity = activity!!
```

Comment:
avoid using the `!!` operator, think of using `safe calls (?.)` instead, but also evaluate if you can receive a nullable activity as a parameter, and in case you can receive it, how are you handling the null state.

read the "checking for null in conditions" section in the kotlin documentation, and you will know why having a `!!` operator is a bad idea in this case.
https://kotlinlang.org/docs/reference/null-safety.html#the--operator
https://kotlinlang.org/docs/reference/null-safety.html#safe-calls

Reply:
done